core/client/fs/src/main/java/alluxio/client/block/AlluxioBlockStore.java

```diff
@@ -214,6 +214,15 @@ public BlockInStream getInStream(BlockInfo info, InStreamOptions options,
     }
 
     try {
+      // ALLUXIO-11172: If the worker is in a container, use the container hostname
+      // to establish the connection.
+      if (!dataSource.getContainerHost().equals("")) {
+        String containerName = dataSource.getContainerHost();
+        LOG.debug("Worker is in a container. Replace host {} with container host {}",
+                dataSource.getHost(), containerName);
+        dataSource.setHost(containerName);
```

Comment:
I suggest not replacing the field here and instead using `containerHost` instead of `host` in https://github.com/Alluxio/alluxio/blob/master/core/common/src/main/java/alluxio/util/network/NetworkAddressUtils.java#L631

 This is used across both read and write paths (`GrpcDataReader` and `GrpcDataWriter`) and will act as the single point of change